Group countries

sverhoeven opened the pull request:
Fixes #51 

On map page
Hover
![Screenshot from 2025-03-25 10-59-20](https://github.com/user-attachments/assets/99a0e1d1-8de9-42e4-bccd-6dd09bef768c)
Dropdown
![Screenshot 2025-03-24 164707](https://github.com/user-attachments/assets/951be693-e326-48de-b99c-e4ae2cf32b9a)

On France page
![Screenshot from 2025-03-25 10-59-58](https://github.com/user-attachments/assets/1e5ea1d3-8fd6-4de2-ace2-592af1634564)
On EU page
![Screenshot from 2025-03-25 11-00-30](https://github.com/user-attachments/assets/d933a7f5-02e9-496b-8cf6-647482ee8f1f)

TODO
- [x] make ws return regions of country and countries of region
- [x] region page lost that regions flag
- [x] works when you do not have all countries netcdf files downloaded
- [x] shows flags for all non-country regions, 
- [x] check all region pages with tests/region.test.ts
- [x] collapse big (>5) list of regions to a drop down

To review:
1. Check that countries_of_regions variable in ws.py has correct data, 
   - regions from XLSX from  https://github.com/pbl-nl/website-carbon-budget-explorer/issues/51#issuecomment-2748289075 have been copied correctly
    - regions not in XLSX are correct: FSM
1. Try to replicate screenshots

sverhoeven committed:
Added countries of region and regions of country to region endpoints

sverhoeven committed:
Have regions and countries section on country popup

sverhoeven committed:
WIP: On region page show country members of region and regions of a country

sverhoeven committed:
Sort regions of country

sverhoeven committed:
Use dropdown to render regions of country and countries of region

sverhoeven committed:
Make work with non all region netcdf files

sverhoeven committed:
on map page on hover then show country name + metric near mouse

sverhoeven committed:
fix eslint error

sverhoeven committed:
Dont list FSM in FSM

sverhoeven committed:
Correct region page tests

sverhoeven committed:
Fix ts error

sverhoeven committed:
Get isEUMemberState from server

Replaces list of eu countries in svelte

sverhoeven committed:
Dont log

Peter9192 commented on src/lib/charts/LeafletMap.svelte:
```diff
@@ -65,21 +67,27 @@
+
+		const feature: RegionFeature = e.sourceTarget.feature;
+		const metric = getMetric(feature);
+		tooltip = {
```
Might be nice to adjust the position of the tooltip on mousemove

sverhoeven replied:
A for me it was less busy that the tooltip stayed in same position when hovering on same country.

Peter9192 commented on ws.py:
```diff
@@ -280,6 +280,207 @@ def build_regions():
+        "Australasia": ["AUS", "NZL"],
+        "Umbrella": ["AUS", "CAN", "ISL", "ISR", "JPN", "KAZ", "NOR", "NZL", "UKR", "USA"],
+        "G7": ["CAN", "DEU", "FRA", "GBR", "ITA", "JPN", "USA"],
+        "G20": ["ARG", "AUS", "BRA", "CAN", "CHN", "DEU", "FRA", "GBR", "IND", "IDN", "ITA", "JPN", "KOR", "MEX", "RUS", "SAU", "TUR", "USA", "ZAF"],
```
G20 includes 19 countries + EU and African group (as groups). This list has 19 entries, but not the countries in those groups. In the excel sheet, members of EU and African group are included.

Peter9192 commented on ws.py:
```diff
@@ -280,6 +280,207 @@ def build_regions():
+            "VUT",
+            "YEM",
+            "ZMB",
+            "ZWE",
```
LDC has 64 entries in this list but only 46 in the excel sheet. E.g. ZWE (Zimbabwe) is present here but not in excel sheet.

Peter9192 commented on ws.py:
```diff
@@ -280,6 +280,207 @@ def build_regions():
+            "ZMB",
+            "ZWE",
+        ],
+        "SIDS": [
```
SIDS are 29 countries in this list but 39 in the excel sheet. E.g TLS (Timor-Leste) is missing in this list.

Peter9192 commented on ws.py:
```diff
@@ -280,6 +280,207 @@ def build_regions():
+        "Umbrella": ["AUS", "CAN", "ISL", "ISR", "JPN", "KAZ", "NOR", "NZL", "UKR", "USA"],
+        "G7": ["CAN", "DEU", "FRA", "GBR", "ITA", "JPN", "USA"],
+        "G20": ["ARG", "AUS", "BRA", "CAN", "CHN", "DEU", "FRA", "GBR", "IND", "IDN", "ITA", "JPN", "KOR", "MEX", "RUS", "SAU", "TUR", "USA", "ZAF"],
+        "FSM": ["MHL", "PLW", "NRU"],
```
I'm not sure if this list is correct. Doesn't seem to match https://www.iso.org/obp/ui/#iso:code:3166:FM

Peter9192 commented on src/routes/map/+page.svelte:
```diff
@@ -98,45 +89,25 @@
 				<div class="relative h-full w-full">
 					<div
-						class="absolute left-0 top-0 z-[500] min-h-[4.5rem] w-64 rounded-br-md bg-white p-2 shadow"
+						class="absolute left-0 top-0 z-[500] flex w-fit flex-row items-center gap-1 rounded-br-md bg-white p-2 shadow"
```
I think we can make this pop out a bit more. Was playing with styles, for example:

```suggestion
						class="absolute left-7 top-7 z-[500] flex w-fit flex-row items-center bg-white p-2 shadow-2xl border border-gray-300 rounded-lg"
```

Peter9192 commented on src/routes/map/+page.svelte:
```diff
@@ -98,45 +89,25 @@
-									</div>
+						<div>Click on a country or</div>
+						<details class="dropdown">
+							<summary class="btn btn-ghost btn-sm font-normal">Select country &#9660;</summary>
```
I think the button looks better (and better aligned) with 

```suggestion
							<summary class="btn btn-ghost btn-sm normal-case">select country &#9660;</summary>
```

sverhoeven replied:
So you want the button to look like the text before it.

Peter9192 replied:
Hmm it looks like my suggestions don't display correctly, at least not on my phone. I can try to see if I still have my edits somewhere. But yes, I want it to look like on the screenshot.

Peter9192 commented on src/lib/RegionList.svelte:
```diff
@@ -10,10 +10,31 @@
 </script>
 
 <!-- TODO add filter by name and filter by non-country (aka ISO2===null) -->
+<h1 class="text-lg">Regions</h1>
```
Is `Regions` the best name to use for 'Non-country regions'? Also for file naming, API, et cetera might be confusing; when does 'region' refer to any region (incl countries) and when does it refer to a group of countries?

Perhaps 'Collectives' or "International Groups" or "Global Alliances"?

sverhoeven replied:
Currently non-country regions are called regions.
Peter has some alternate naming suggestion: 'Collectives' or "International Groups" or "Global Alliances"
Some more suggestions 'Geopolitical entities', 'Regional blocs', 'Supranational ...', ' Intergovernmental  ...'

@MarkMDekker what is your preference?

Peter9192 replied:
Let's keep this as is for now and possibly improve later on. Don't want to hold up the merge.

Peter9192 reviewed (changes requested):
All works well, I can reproduce the behaviour. Wonder if 'region' terminology is not too confusing.

I checked the groups but there seem to be some inconsistencies between excel sheet and website

I played a little bit with the styling to make the select box stand out more from the page (see inline suggestions): 
![image](https://github.com/user-attachments/assets/2c1e5ba4-bdcf-4022-9534-cb3fa56ceb96)

sverhoeven committed:
Get groups from xlsx

```python
import pandas as pd
fn = 'data/Version_0_4_2/UNFCCC_Parties_Groups_noeu.xlsx'
df = pd.read_excel(fn)
c = df.dropna(subset=['G20'])["Country ISO Code"].values.tolist()
sorted(c)
```

sverhoeven committed:
Make region list trigger flow better + less columns on region list on small screens

sverhoeven commented:
I have updated the region members. I think some AI slipped in, now used

```python
import pandas as pd
fn = 'data/Version_0_4_2/UNFCCC_Parties_Groups_noeu.xlsx'
df = pd.read_excel(fn)
c = df.dropna(subset=['G20'])["Country ISO Code"].values.tolist()
sorted(c)
```

Also the 
![localhost_5173_map_allocTime=2030](https://github.com/user-attachments/assets/136715b7-0a11-4123-b935-b7e1649b7526)
region list popup button text has been changed

sverhoeven committed:
Merge remote-tracking branch 'origin/staging' into group-countries-mouse-tooltip
